jenkins/Jenkinsfile

```diff
@@ -0,0 +1,321 @@
+                      python -m ensurepip --upgrade
+                      python -m pip install --upgrade pip
+                      python -m pip install --upgrade virtualenv
+
+                      virtualenv .venv
+                      source .venv/bin/activate
+
+                      pip install cloudify==5.1.0
+                      cfy profile use ${env.OS_IP} -u ${env.OS_USERNAME} -p ${env.OS_PASSWORD} -t ${env.OS_TENANT}
+                      pushd 'bp'
+                        cfy install -b ec2-manager-install-blueprint-${env.BUILD_NUMBER} ec2-manager-install-blueprint.yaml
+                      popd
+                      cfy deployments capabilities ec2-manager-install-blueprint-${env.BUILD_NUMBER} --json > capabilities.json
+                      echo -e \$(cat capabilities.json | jq '.key_content.value' | tr -d '"') > ~/.ssh/ec2_ssh_key && chmod 600 ~/.ssh/ec2_ssh_key
+                      sleep 160
+                      ssh-keyscan -H \$(cat capabilities.json | jq '.endpoint.value' | tr -d '"') >> ~/.ssh/known_hosts
+                      echo 'ClientAliveInterval 50' | sudo tee --append /etc/ssh/sshd_config
+                      """, label:'Configure and install blueprint on manager'
+                    }
+                  }
+                }
+              }
+            }
+          }             
+        }
+      }
+    }
+        stage('install_manager') {
+          environment {
+            CONTAINER_NAME = "cfy_aio"
+            IMAGE_NAME = "cfy_manager_image"
+          }
+          steps {
+            catchError(message: 'Failure on: manager-install install_manager', buildResult: 'SUCCESS', stageResult: 'FAILURE') {
+              container('py27') {
+                dir("${env.WORKSPACE}/${env.PROJECT}/jenkins") {
+                  withVault([configuration: configuration, vaultSecrets: secrets]){
+                    sh script: """#!/bin/bash
+                    ssh -i ~/.ssh/ec2_ssh_key -l centos \$(cat capabilities.json | jq '.endpoint.value' | tr -d '"') /bin/bash << 'EOT'
+sudo yum -y update
+sudo service docker start
+git clone https://github.com/cloudify-cosmo/cloudify-manager-install.git && cd cloudify-manager-install && git checkout ${env.BRANCH_NAME}
+docker run --net host -dt -v ${pwd}:/mount python python -m http.server -d /mount
+echo 'Build manager container'
+set -eux
+ls -la .
+pushd packaging/docker
+  docker build --network host --build-arg rpm_file=https://cloudify-release-eu.s3.amazonaws.com/cloudify/${env.S3_BUILD_PATH}/cloudify-manager-install-${env.VERSION}-${env.PRERELEASE}.el7.centos.x86_64.rpm --tag ${env.IMAGE_NAME} .
+popd
+echo 'Run manager container'
+set -eux
+docker run --name ${env.CONTAINER_NAME} -d -v /sys/fs/cgroup:/sys/fs/cgroup:ro --tmpfs /run --tmpfs /run/lock ${env.IMAGE_NAME}
+echo "Waiting for ${env.CONTAINER_NAME} to start"
+docker exec ${env.CONTAINER_NAME} cfy_manager wait-for-starter
+echo 'Check Manager status'
+sleep 20
```

Comment:
why sleep? does wait-for-starter not enough?

Reply:
without sleep or 'wait-for-starter' it might not be ready in time for the status validation.
Besides that, Lukasz recommended on that